pa4/procedures.sql

```diff
@@ -0,0 +1,38 @@
+-- HW4
+
+DELIMITER $$
+CREATE PROCEDURE product_details_by_name(IN product_name VARCHAR(255))
+BEGIN
+SELECT name, description, quantity, cost FROM product WHERE name = product_name;
+END $$
+
+CREATE PROCEDURE total_income_from_payments(OUT total INT)
+BEGIN
+SELECT SUM(amount) INTO total FROM payment;
+END $$
+
+CREATE PROCEDURE update_total_income_by_dates(INOUT total_income INT, IN start_date DATE, IN end_date DATE)
+BEGIN
+    DECLARE sales INT DEFAULT 0;
+SELECT SUM(amount) INTO sales FROM payment WHERE date BETWEEN start_date AND end_date;
+SET total_income = total_income + sales;
+END $$
+
+CREATE PROCEDURE update_product_quantity(IN prod_id INT, IN quantity_minus INT)
+BEGIN
+    DECLARE new_quantity INT;
+
+START TRANSACTION;
```

Comment:
Please use tabulation for all procedures for all statements: BEGIN-END/START TRANSACTION, etc.